extension/src/paymentHandler/payment-handler.js

```diff
@@ -20,13 +23,35 @@ const PAYMENT_METHOD_TYPE_KLARNA_METHODS = [
   'klarna_account',
 ]
 
-async function handlePayment(paymentObject) {
+async function handlePayment(paymentObject, authToken) {
   if (!_isAdyenPayment(paymentObject))
     // if it's not adyen payment, ignore the payment
     return { success: true, data: null }
 
-  const paymentValidator = ValidatorBuilder.withPayment(paymentObject)
+  let paymentValidator = ValidatorBuilder.withPayment(paymentObject)
     .validateMetadataFields()
+  if (paymentValidator.hasErrors())
+    return {
+      success: false,
+      data: paymentValidator.buildCtpErrorResponse(),
+    }
+
+  const ctpProjectKey = paymentObject.custom.fields.commercetoolsProjectKey
+  if (!auth.hasValidAuthorizationHeader(ctpProjectKey, authToken)) {
+    return {
+      success: false,
+      data: {
+        errors: [
+          {
+            code: 'InvalidOperation',
```

Comment:
Why is it invalid operation? Why not code: `401` and message: `Unauthorized`?

Reply:
https://github.com/commercetools/commercetools-adyen-integration/pull/584/commits/168b68b68314b591a950d3b6ad3d1fdae63df9af

I have tested in e2e test, all defined error code is categorized into 400 http status.
https://docs.commercetools.com/api/projects/api-extensions#validation-failed